test/config-luatest/config_test.lua

```diff
@@ -343,3 +343,124 @@ g.test_remaining_vinyl_options = function()
         t.assert_equals(box.cfg.vinyl_timeout, 7.5)
     end)
 end
+
+g.test_bootstrap_leader = function(g)
+    local dir = treegen.prepare_directory(g, {}, {})
+    local config = [[
+        credentials:
+          users:
+            guest:
+              roles:
+              - super
+
+        iproto:
+          listen: unix/:./{{ instance_name }}.iproto
+
+        groups:
+          group-001:
+            replicasets:
+              replicaset-001:
+                bootstrap_leader: 'instance-002'
```

Comment:
Nit: The error message depends on the order of the checks in the code, because this configuration is incorrect in two ways: no `bootstrap_strategy`, non-existing `bootstrap_leader`. I would change the leader to `instance-001`.

Reply:
Fixed.